src/mg/PAMI/Message/IncomingMessage.php

```diff
@@ -81,6 +87,33 @@ public function getRawContent()
     }
 
     /**
+     * Returns the channel variables for all reported channels.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * The channel names will be lowercased.
+     *
+     * @return array
+     */
+    public function getAllChannelVariables()
+    {
+      return $this->channelVariables;
+    }
+
+    /**
+     * Returns the channel variables for the given channel.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * @param string $channel Channel name. If not given, will return variables
+     * for the "current" channel.
+     *
+     * @return array
+     */
+    public function getChannelVariables($channel = 'default')
```

Comment:
I wouldn't trust asterisk to always set the channel name, and also having a default way of getting the variables allows one to get the channel variables without knowing in advance the channel name for that event (and would work even if they are not sent, AMI is pretty inconsistent sometimes).

About the bug.. I don't see it. The test actually asserts that all the variables are returned. Could you write a test case that fails with the given code?